core/src/test/scala/zio/config/RecursiveConfigTest.scala

```diff
@@ -110,4 +169,31 @@ object RecursiveConfigTestUtils {
 
   val recursiveValue: Data = Data(Row(1, Some(Data(Row(2, None)))))
 
+  sealed trait Expr
+  case class Lit(n: Int)            extends Expr
+  case class Add(items: List[Expr]) extends Expr
+
+  def expr: ConfigDescriptor[Expr] = {
+    val lit: ConfigDescriptor[Expr] = int.xmapEither(
+      n => Right(Lit(n)), {
+        case Lit(n) => Right(n)
+        case _      => Left(s"Not Lit")
+      }
+    )
+
+    val add: ConfigDescriptor[Expr] = nested("add")(
+      list(expr)
+        .apply(
+          lst => Add(lst), {
+            case Add(items) => Some(items)
+            case _          => None
+          }
+        )
+    )
+
+    lit <> add
+  }
+
+  val exprValue  = Add(List(Lit(1), Add(List(Add(List(Lit(2), Lit(3))), Lit(4))), Lit(5)))
+  val exprSource = ConfigSource.fromPropertyTree(write(expr, exprValue).toOption.get, "test", LeafForSequence.Invalid)
```

Comment:
@vigoo  Will this fix the issue in your zio-aws docs?

Reply:
I don't think so, will look into that when I have time.